ql/src/java/org/apache/hadoop/hive/ql/stats/BasicStatsTask.java

```diff
@@ -512,6 +513,8 @@ private String toString(Map<String, String> parameters) {
         if (dpPartSpecs != null) {
           // load the list of DP partitions and return the list of partition specs
           list.addAll(dpPartSpecs);
+          // Reload partition metadata because another BasicStatsTask instance may have updated the stats.
+          list = db.getPartitionsByNames(table, list.stream().map(Partition::getName).collect(Collectors.toList()));
```

Comment:
Maybe it would be slightly more readable like below:
```suggestion
          // Reload partition metadata because another BasicStatsTask instance may have updated the stats.
          List<String> partNames = dpPartSpecs.stream().map(Partition::getName).collect(Collectors.toList());
          list.addAll(db.getPartitionsByNames(table, partNames));
```